pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -2290,7 +2290,7 @@ private void createPendingLoadTopic() {
     public CompletableFuture<PartitionedTopicMetadata> fetchPartitionedTopicMetadataAsync(TopicName topicName) {
         // gets the number of partitions from the configuration cache
         return pulsar.getPulsarResources().getNamespaceResources().getPartitionedTopicResources()
-                .getAsync(partitionedTopicPath(topicName)).thenApply(metadata -> {
```

Comment:
The client will fetch the topic metadata for each 60s by default, this will introduce more workload on the Zookeeper if you have many clients in your cluster.

Reply:
So should we invalidate the path at the end of the delete operation instead of replacing with an Optional.empty() ?